packages/wallets/src/components/FlowField/FlowTextField.tsx

```diff
@@ -1,34 +1,68 @@
-import React, { forwardRef, Ref } from 'react';
+import React, { forwardRef, Ref, useEffect, useState } from 'react';
 import { Field, FieldProps } from 'formik';
+import * as Yup from 'yup';
 import WalletTextField, { WalletTextFieldProps } from '../Base/WalletTextField/WalletTextField';
+import { useFlow } from '../FlowProvider';
 
 export interface TFlowFieldProps extends WalletTextFieldProps {
     isInvalid?: WalletTextFieldProps['isInvalid'];
     name: string;
+    validationSchema?: Yup.AnySchema;
 }
 
 /**
  * This component is just a wrapper to the Field Formik component and WalletTextField
  * Use this component when you are using the FlowProvider with a form and several inputs,
  * and you want those input values to be tracked and validated
  */
-const FlowTextField = forwardRef(({ isInvalid, name, ...rest }: TFlowFieldProps, ref: Ref<HTMLInputElement>) => {
-    return (
-        <Field name={name}>
-            {({ field, form, meta }: FieldProps) => {
-                return (
-                    <WalletTextField
-                        {...rest}
-                        {...field}
-                        errorMessage={form.errors[name]}
-                        isInvalid={isInvalid || (meta.value !== meta.initialValue && Boolean(form.errors[name]))}
-                        ref={ref}
-                    />
-                );
-            }}
-        </Field>
-    );
-});
+const FlowTextField = forwardRef(
+    (
+        { defaultValue, errorMessage, isInvalid, name, validationSchema, ...rest }: TFlowFieldProps,
+        ref: Ref<HTMLInputElement>
+    ) => {
+        const [hasTouched, setHasTouched] = useState(false);
+        const { setFormValues } = useFlow();
+
+        const validateField = (value: unknown) => {
+            try {
+                if (validationSchema) {
+                    validationSchema.validateSync(value);
+                }
+            } catch (err: unknown) {
+                return (err as Yup.ValidationError).message;
+            }
+        };
+
+        useEffect(() => {
+            if (defaultValue) {
+                setFormValues(name, defaultValue);
+            }
```

Comment:
We need to sync the Formik state if we are passing a default value to the text field, otherwise the initial value provided by Formik will be empty